backend/services/deployments/app/app.go

```diff
@@ -1245,6 +1247,50 @@ func (d *Deployments) getDeploymentGroups(
 	return groups, nil
 }
 
+func getDeploymentFilter(
+	constructor *model.DeploymentConstructor,
+) *model.Filter {
+
+	var filter *model.Filter
+
+	if len(constructor.Group) > 0 {
+		filter = &model.Filter{
+			Terms: []model.FilterPredicate{
+				{
+					Scope:     InventoryGroupScope,
+					Attribute: InventoryGroupAttributeName,
+					Type:      "$eq",
+					Value:     constructor.Group,
+				},
+			},
+		}
+	} else if constructor.AllDevices {
+		filter = &model.Filter{
+			Terms: []model.FilterPredicate{
+				{
+					Scope:     InventoryIdentityScope,
+					Attribute: InventoryStatusAttributeName,
+					Type:      "$eq",
+					Value:     InventoryStatusAccepted,
+				},
+			},
+		}
+	} else if len(constructor.Devices) > 0 {
+		filter = &model.Filter{
+			Terms: []model.FilterPredicate{
+				{
+					Scope:     InventoryIdentityScope,
+					Attribute: InventoryIdAttributeName,
+					Type:      "$in",
```

Comment:
This is not a valid filtering operator according to inventory, should we add it? https://github.com/mendersoftware/mender-server/blob/f9ce3d325be98697ab7cee464cbbcba6d5683b1c/backend/services/inventory/model/filters.go#L21-L24

Reply:
@alfrunes I created a follow up ticket - https://northerntech.atlassian.net/browse/MEN-7667.
With this PR we don't execute the filter. Filter will be executed later by the UI (there will be separate UI task to start using this functionality), so it will be blocked by the inventory one.